lib/datadog/core/telemetry/request.rb

```diff
@@ -31,6 +31,12 @@ def build_payload(event, seq_id)
 
           def application
             config = Datadog.configuration
+
+            tracer_version = Core::Environment::Identity.gem_datadog_version_semver2
+            if config.respond_to?(:ci) && config.ci.enabled && defined?(::Datadog::CI::VERSION)
+              tracer_version = "#{tracer_version}+ci-#{::Datadog::CI::VERSION}"
+            end
```

Comment:
Really minor: Consider starting with the `defined?` first? I suspect that one is cheaper than any of the other checks, and will allows us to bail out earlier.